WordPress/openverse-frontend
# Add import extension linting rule

## Fixes


Fixes #1138  by @sarayourfriend 

## Description


Kudos to @dhruvkb for bringing this up.

This PR enables the `import/extensions` rule and fixes all errors that were raised by it. It also fixes (silences with explanation) the `vue/no-v-html` warning that we've all be secretly ignoring for the past week or so :speak_no_evil: 

Are there any other file extensions that should be exempted from this rule? I thought about adding `tsx` but we shouldn't need to use `tsx` as far as I know, right? If we do then we can add it then (I imagine it'd be an explicit decision at that point?)

## Testing Instructions

Checkout the branch and add an import with a `js` or `ts` extension and confirm that it raises an error. Also, import a `vue` file without the extension and confirm that it produces an error.

## Checklist

- [N/A] I added or updated tests for the changes I made (if applicable).
- [N/A] I added or updated documentation (if applicable).

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
